frontend/pages/hosts/details/cards/HostSummary/HostSummary.tsx

```diff
@@ -108,47 +178,40 @@ const HostSummary = ({
   renderActionButtons,
   deviceUser,
   osSettings,
+  hostMdmDeviceStatus,
 }: IHostSummaryProps): JSX.Element => {
   const { status, platform } = titleData;
 
   const renderRefetch = () => {
     const isOnline = titleData.status === "online";
+    let isDisabled = false;
+    let tooltip: React.ReactNode = <></>;
+
+    // deviceStatus can be `undefined` in the case of the MyDevice Page not sending
+    // this prop. When this is the case or when it is `unlocked`, we only take
+    // into account the host being online or offline for correctly render the
+    // refresh button. If we another value for deviceStatus, we take it account
+    // as well.
+    if (
+      hostMdmDeviceStatus === undefined ||
+      hostMdmDeviceStatus === "unlocked"
+    ) {
+      isDisabled = !isOnline;
+      tooltip = !isOnline ? REFETCH_TOOLTIP_MESSAGES.offline : null;
+    } else {
+      isDisabled = true;
+      tooltip = !isOnline
+        ? REFETCH_TOOLTIP_MESSAGES.offline
+        : REFETCH_TOOLTIP_MESSAGES[hostMdmDeviceStatus];
+    }
 
     return (
-      <>
-        <div
-          className="refetch"
-          data-tip
-          data-for="refetch-tooltip"
-          data-tip-disable={isOnline || showRefetchSpinner}
-        >
-          <Button
-            className={`
-            button
-            ${!isOnline ? "refetch-offline tooltip" : ""}
-              ${showRefetchSpinner ? "refetch-spinner" : "refetch-btn"}
-            `}
-            disabled={!isOnline}
-            onClick={onRefetchHost}
-            variant="text-icon"
-          >
-            <Icon name="refresh" color="core-fleet-blue" size="small" />
-            {showRefetchSpinner
-              ? "Fetching fresh vitals...this may take a moment"
-              : "Refetch"}
-          </Button>
-        </div>
-        <ReactTooltip
-          place="top"
-          effect="solid"
-          id="refetch-tooltip"
-          backgroundColor={COLORS["tooltip-bg"]}
-        >
-          <span className={`${baseClass}__tooltip-text`}>
-            You can’t fetch data from <br /> an offline host.
-          </span>
-        </ReactTooltip>
-      </>
+      <RefreshButton
+        isDisabled={isDisabled}
+        isFetching={showRefetchSpinner}
```

Comment:
nice cleanup

Reply:
thanks. I had to change the functionality and it was getting a bit more complicated so i decided to pull it out and update.